test/linear_combination.jl

```diff
@@ -82,14 +88,21 @@
         end
 
         @testset "dims" begin
+            @testset "dims = :" begin
+                d = Colon()
+                @test_throws ArgumentError Transforms.apply(A, lc; dims=d)
+            end
+
             @testset "dims = 1" begin
```

Comment:
does this also work if passing in the axis key? e.g. `foo`, `bar`?

Reply:
I didn't check it with `Power` so if it doesn't work don't worry about it

Reply:
I doubt it

Reply:
How do you do the same with AxisArrays though? If I use `(Colon(), :foo, :bar)` like for KeyedArray in `test/power.jl` I get

```
dims = foo: Error During Test at /Users/bencottier/JuliaEnvs/Transform.jl/test/power.jl:65
  Got exception outside of a @test
  MethodError: no method matching length(::Symbol)
  Closest candidates are:
    length(!Matched::Base.Iterators.Flatten{Tuple{}}) at iterators.jl:1061
    length(!Matched::Base.MethodList) at reflection.jl:872
    length(!Matched::DataStructures.EnumerateAll) at /Users/bencottier/.julia/packages/DataStructures/ixwFs/src/multi_dict.jl:96
    ...
  Stacktrace:
   [1] #eachslice#196 at ./abstractarraymath.jl:496 [inlined]
   [2] apply!(::AxisArray{Int64,2,Array{Int64,2},Tuple{Axis{:foo,Array{String,1}},Axis{:bar,Array{String,1}}}}, ::Power; dims::Symbol, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/bencottier/JuliaEnvs/Transform.jl/src/transformers.jl:57
   [3] apply(::AxisArray{Int64,2,Array{Int64,2},Tuple{Axis{:foo,Array{String,1}},Axis{:bar,Array{String,1}}}}, ::Power; kwargs::Base.Iterators.Pairs{Symbol,Symbol,Tuple{Symbol},NamedTuple{(:dims,),Tuple{Symbol}}}) at /Users/bencottier/JuliaEnvs/Transform.jl/src/transformers.jl:64
   [4] macro expansion at /Users/bencottier/JuliaEnvs/Transform.jl/test/power.jl:66 [inlined]
   [5] macro expansion at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Test/src/Test.jl:1190 [inlined]
   [6] macro expansion at /Users/bencottier/JuliaEnvs/Transform.jl/test/power.jl:65 [inlined]
   [7] macro expansion at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Test/src/Test.jl:1115 [inlined]
   [8] macro expansion at /Users/bencottier/JuliaEnvs/Transform.jl/test/power.jl:62 [inlined]
   [9] macro expansion at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Test/src/Test.jl:1115 [inlined]
   [10] top-level scope at /Users/bencottier/JuliaEnvs/Transform.jl/test/power.jl:3
   [11] include(::String) at ./client.jl:457
   [12] top-level scope at /Users/bencottier/JuliaEnvs/Transform.jl/test/runtests.jl:12
   [13] top-level scope at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Test/src/Test.jl:1115
   [14] top-level scope at /Users/bencottier/JuliaEnvs/Transform.jl/test/runtests.jl:11
   [15] include(::String) at ./client.jl:457
   [16] top-level scope at none:6
   [17] eval(::Module, ::Any) at ./boot.jl:331
   [18] exec_options(::Base.JLOptions) at ./client.jl:272
   [19] _start() at ./client.jl:506
```

Reply:
it looks like `eachslice` doesn't work on AxisArrays maybe. Nevermind it was only a passing thought.

Reply:
yeah it doesn't work without some extra voodoo

```julia
julia> eachslice(A, dims=axisdim(A, Axis{:foo}))
Base.Generator{Base.OneTo{Int64},Base.var"#199#202"{AxisArray{Int64,2,Array{Int64,2},Tuple{Axis{:foo,Array{String,1}},Axis{:bar,Array{String,1}}}},Tuple{},Tuple{Colon}}}(Base.var"#199#202"{AxisArray{Int64,2,Array{Int64,2},Tuple{Axis{:foo,Array{String,1}},Axis{:bar,Array{String,1}}}},Tuple{},Tuple{Colon}}([1 2 3; 4 5 6], (), (Colon(),)), Base.OneTo(2))
```

And it's been on their todo list for 5 years 🙄 
https://github.com/JuliaArrays/AxisArrays.jl/issues/7

Reply:
I can make this work if we want to, by checking if it's an AxisArray. I might prefer to do this in a follow up MR just to keep diffs smaller.

Reply:
https://github.com/invenia/Transforms.jl/issues/13